app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,373 @@
+    }
+
+    fn is_state_switch(&self, model: &Model) -> bool {
+        match model {
+            Model::Text(_) => self.state.get() != State::Text,
+            Model::Separator => self.state.get() != State::Separator,
+            Model::Ellipsis => self.state.get() != State::Ellipsis,
+        }
+    }
+
+    fn width(&self, text_offset: f32) -> f32 {
+        match self.state.get() {
+            State::Text => self.text.width.value() + text_offset * 2.0,
+            State::Separator => separator::ICON_WIDTH,
+            State::Ellipsis => ellipsis::ICON_WIDTH,
+        }
+    }
+}
+
+// === Params ===
+
+/// The parameters of Breadcrumbs' entries.
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Default)]
+pub struct Params {
+    pub margin:                   f32,
+    pub hover_color:              color::Rgba,
+    pub font:                     ImString,
+    pub text_offset:              f32,
+    pub text_size:                text::Size,
+    pub selected_color:           color::Rgba,
+    pub highlight_corners_radius: f32,
+    pub greyed_out_color:         color::Rgba,
+    pub greyed_out_start:         Option<usize>,
+}
+
+impl Params {
+    pub(crate) fn from_style(
+        style: &StyleWatchFrp,
+        network: &frp::Network,
+        init: frp::Source<()>,
+    ) -> frp::Sampler<Self> {
+        frp::extend! { network
+            let margin = style.get_number(theme::entry::margin);
+            let hover_color = style.get_color(theme::entry::hover_color);
+            let font = style.get_text(theme::entry::font);
+            let text_offset = style.get_number(theme::entry::text_offset);
+            let text_size = style.get_number(theme::entry::text_size);
+            let selected_color = style.get_color(theme::entry::selected_color);
+            let highlight_corners_radius = style.get_number(theme::entry::highlight_corners_radius);
+            let greyed_out_color = style.get_color(theme::entry::greyed_out_color);
+            greyed_out_start <- init.constant(None);
+            text_params <- all4(&init, &text_offset,&text_size,&font);
+            colors <- all4(&init, &hover_color,&selected_color,&greyed_out_color);
+            params <- all_with6(&init,&margin,&text_params,&colors,&highlight_corners_radius,
+                &greyed_out_start,
+                |_,&margin,text_params,colors,&highlight_corners_radius,&greyed_out_start| {
+                    let (_, text_offset,text_size,font) = text_params;
+                    let (_, hover_color,selected_color,greyed_out_color) = colors;
+                    Params {
+                        margin,
+                        text_offset: *text_offset,
+                        text_size: text::Size::from(*text_size),
+                        hover_color:*hover_color,
+                        font: ImString::new(font),
+                        selected_color: *selected_color,
+                        highlight_corners_radius,
```

Comment:
This code extends frp network not belonging to thsi struct. Unfortunately, due to lack of proper docs, I dont understand what this structure is for. Anyway, what network does it extend? Please remember that after you extend FRP network, you CAN'T remove this structure leving the FRP network - otherwise you will cause memory leak. Are you 100% sure that when this struct is dropped the network is dropped as well? If you are not sure, you can't do it and you'd need to use FRP Bridge Network (used for example in the nodes implementation), which will be automatically dropped OR create a local network in this struct. Anyway, this is not documented. If you are 100% sure that network will be droped with thsi struct, this needs to be extremely well documented and explained why it happens this way.

Reply:
There should be no mistake here. To avoid confusion, I moved the function closer to its usage (to the root of the crate) and added docs. Keep in mind that the return value is `Sampler<Self>`, not `Self`, and the Sampler is later used in the network we extended here.

Reply:
Good to know that. Still, it sohuld always be well documented because it's dangereous. In new FRP implementation we will disallow `extend` function completely. Thanks for moving it closer to the usage point and the docs.